timemachine/cpp/src/barostat.cu

```diff
@@ -42,18 +42,29 @@ MonteCarloBarostat::MonteCarloBarostat(
         std::cout << "warning pressure more than 10bar" << std::endl;
     }
 
-    mt_ = std::mt19937(seed_);
-    dist_ = std::uniform_real_distribution<double>(0.0, 1.0);
-
-    //gpuErrchk(cudaMalloc(&d_group_idxs_, group_idxs_.size()*sizeof(*d_group_idxs_)));
+    curandErrchk(curandCreateGenerator(&cr_rng_, CURAND_RNG_PSEUDO_DEFAULT));
+    gpuErrchk(cudaMalloc(&d_rand_, 2*sizeof(double)));
```

Comment:
As an optimization later on, it may be more efficiently to generate RNGs with a bigger buffer, and I assume you've profiled to verify that this isn't a major contribution to the slow down.

Reply:
I have verified that this is not a contribution to the slow down. The neighborlist continues to appear as the bottle neck, taking up 30% of the computation time as the barostat interval approaches 1